### DISCUSSION TITLE
Add the ability to assign hotkeys to individual Spotlight Actions

### DISCUSSION THREAD

**Discussion by dipique** [mantinedev/mantine#1664] [Feature requests]
Because spotlight is already registering actions and hotkeys, it would be very helpful if hotkeys could also be registered to trigger specific actions. This would currently be accomplished by manually registering hotkeys in your app and then triggering spotlight actions.

### Unsolicited implementation ideas

`interface SpotlightAction` could have the following added: 

```
/** Keyboard shortcut that triggers this specific action without opening the spotlight overlay */
  shortcut?: string;
```

The shortcut parsing could follow the same rules as the `shortcut` prop for `SpotlightProvider`. Based on some cursory investigation, the existing `useHotkeys` hook looks completely suitable for this use.

I am thinking that perhaps the hotkey could be registered inside [prepareAction](https://github.com/mantinedev/mantine/blob/c2f111d85d23603b3197762906841b62093475e2/src/mantine-spotlight/src/use-actions-state/use-actions-state.ts#L5), but my understanding of hooks is very weak.

**Comment by rtivital** [comment]
You can use `use-hotkeys` hook to perform any actions that will be invoked without spotlight being opened

**Reply by dipique** [reply]
Thanks, @rtivital. I believe I mentioned that in the request. The reason I thought this convenience feature was compelling is because hotkey binding is already built into the spotlight component, and mimicking that format at the action level seems like a great convenience feature that leverages already-existing features in an intuitive way.

However, if that doesn't add sufficient value in your opinion, feel free to close the request.
